client/verta/verta/_internal_utils/model_dependencies.py

```diff
@@ -25,19 +25,21 @@ def unwrap(func: Callable) -> Callable:
 
 
 def modules_in_function_body(func: Callable) -> Set[str]:
-    """Return a set of all base modules called within the body of the provided function."""
+    """Return a set of all base modules called within the body of the provided function.
+    The `unbound` value returned by `inspect.getclosurevars` returns a set of names
+    referenced in the function that could not be resolved.
+    """
     _func = unwrap(func)
-    _globals = [
-        value.__name__.split('.')[0]  # strip off submodules and classes
-        for key, value in inspect.getclosurevars(_func).globals.items()
-        if isinstance(value, ModuleType)
-    ]
-    _non_locals = [
-        value.__name__.split('.')[0]  # strip off submodules and classes
-        for key, value in inspect.getclosurevars(_func).nonlocals.items()
-        if isinstance(value, ModuleType)
+    closure_vars: List[Dict[str, Any]] = [
+        val
+        for key, val in inspect.getclosurevars(_func)._asdict().items()
+        if key != "unbound"
     ]
-    return set(_globals + _non_locals)
+    return {
+        inspect.getmodule(val).__name__.split(".")[0]
```

Comment:
Could you rewrite this section to be clearer? I have truly no idea what's happening here(: maybe taking these lines out of the return and adding some variable names could help?

Reply:
Any time I start nesting inside of a comprehension, I ask myself - "Should I really be doing this?"  The answer is usually no.  I dropped that in favor of a clearer, more readable process.  Let me know what you think.